### PR TITLE
Migrate away from `SIM_register_typed_attribute` -- SIMICS-22406

### PR DESCRIPTION
**Pull request by lwaern-intel**
Draft as we need to deal with the compatibility question of the drop of support of dictionary (`D`) types.

### CODE REVIEW

**Review by lwaern-intel — COMMENTED** [review]
Doh, I forgot to make sure getter/setter trampolines work.

**Comment by syssimics** [comment]
Verification [#13471519](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=13471519): ✅ pass

**Comment by syssimics** [comment]
Verification [#13471492](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=13471492): ✅ pass

**Comment by lwaern-intel** [comment]
@mandolaerik nag

File: test/1.4/legacy/legacy_attributes.dml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+attribute a is uint64_attr;
+
+bank b {
+    saved int s;
+    hook() h;
+    attribute a is uint64_attr;
+}
+
+method init() default {
+    local attr_value_t val = SIM_make_attr_uint64(0xC0FFEE);
```
Comment (mandolaerik) [inline test/1.4/legacy/legacy_attributes.dml:21]: Note: we usually use lowercase hex. Unsure if the style mandates it, and for bogus test values I don't care.

**Review by mandolaerik — APPROVED** [review]

File: test/1.4/legacy/legacy_attributes.dml (hunk 1 above)
Comment (lwaern-intel) [reply test/1.4/legacy/legacy_attributes.dml:21]: huh? we do? if so, we are **definitely** not consistent about it like, at all.

**Pull request merged**